airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlSource.java

```diff
@@ -232,20 +232,19 @@ public AirbyteCatalog discover(final JsonNode config) {
 
   @Override
   public List<TableInfo<CommonField<JDBCType>>> discoverInternal(final JdbcDatabase database) throws Exception {
-    final List<TableInfo<CommonField<JDBCType>>> internals = super.discoverInternal(database);
     if (schemas != null && !schemas.isEmpty()) {
-      // process explicitly filtered (from UI) schemas
-      final List<TableInfo<CommonField<JDBCType>>> resultInternals = internals
-          .stream()
-          .filter(this::isTableInRequestedSchema)
-          .toList();
-      for (final TableInfo<CommonField<JDBCType>> info : resultInternals) {
-        LOGGER.debug("Found table (schema: {}): {}", info.getNameSpace(), info.getName());
-      }
-      return resultInternals;
+      return schemas.stream().flatMap(schema -> {
+        LOGGER.info("Get columns for schema: {}", schema);
+        try {
+          return super.discoverInternal(database, schema).stream();
+        } catch (Exception e) {
+          LOGGER.error("Error getting columns for schema: {}", schema, e);
+          return Stream.empty();
```

Comment:
I believe it's OK to re-throw `e` wrapped in a `ConfigErrorException` here. If the connector configuration doesn't even allow listing the columns inside a schema, there's no point in proceeding further. The user should either fix the config by removing the offending schema or fix the connection credentials or grant them more privileges.

I may be wrong. Thoughts @airbytehq/dbsources ?